fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java

```diff
@@ -0,0 +1,493 @@
+                hiveTableName = properties.get(BITMAP_DATA_PROPERTY);
+            }
+        }
+    }
+
+    @Override
+    protected void unprotectedExecuteJob() throws LoadException {
+        LoadTask task = new SparkLoadPendingTask(this, fileGroupAggInfo.getAggKeyToFileGroups(),
+                                                 etlClusterDesc);
+        task.init();
+        idToTasks.put(task.getSignature(), task);
+        Catalog.getCurrentCatalog().getLoadTaskScheduler().submit(task);
+    }
+
+    @Override
+    public void onTaskFinished(TaskAttachment attachment) {
+        if (attachment instanceof SparkPendingTaskAttachment) {
+            onPendingTaskFinished((SparkPendingTaskAttachment) attachment);
+        }
+    }
+
+    private void onPendingTaskFinished(SparkPendingTaskAttachment attachment) {
+        writeLock();
+        try {
+            // check if job has been cancelled
+            if (isTxnDone()) {
+                LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                                 .add("state", state)
+                                 .add("error_msg", "this task will be ignored when job is: " + state)
+                                 .build());
+                return;
+            }
+
+            if (finishedTaskIds.contains(attachment.getTaskId())) {
+                LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                                 .add("task_id", attachment.getTaskId())
+                                 .add("error_msg", "this is a duplicated callback of pending task "
+                                         + "when broker already has loading task")
+                                 .build());
+                return;
+            }
+
+            // add task id into finishedTaskIds
+            finishedTaskIds.add(attachment.getTaskId());
+
+            sparkAppHandle = attachment.getHandle();
+            etlOutputPath = attachment.getOutputPath();
+
+            unprotectedUpdateState(JobState.ETL);
+        } finally {
+            writeUnlock();
+        }
+    }
+
+    @Override
+    protected void unprotectedUpdateState(JobState jobState) {
+        super.unprotectedUpdateState(jobState);
+
+        if (jobState == JobState.ETL) {
+            executeEtl();
+        }
+    }
+
+    // update etl time and state in spark load job
+    private void executeEtl() {
+        etlStartTimestamp = System.currentTimeMillis();
+        state = JobState.ETL;
+    }
+
+    public void updateEtlStatus() throws Exception {
+        if (state != JobState.ETL) {
+            return;
+        }
+
+        // get etl status
+        Preconditions.checkNotNull(sparkAppHandle);
+        SparkEtlJobHandler handler = new SparkEtlJobHandler();
+        EtlStatus status = handler.getEtlJobStatus(sparkAppHandle, id,
+                                                   etlClusterDesc.getProperties().get("spark.status_server"));
+        switch (status.getState()) {
```

Comment:
I think we can print doris job id and spark appid here to connect doris's job to spark app.